plugins/inputs/prometheus/target_allocator.go

```diff
@@ -0,0 +1,197 @@
+		reloadConfigHandler: nil,
+	}
+	err := tam.loadConfig(filename)
+	if err != nil {
+		level.Warn(logger).Log("msg", "Could not load config for target allocator from file", "filename", filename, "err", err)
+		return &tam
+	}
+	if tam.config == nil {
+		return &tam
+	}
+	tam.enabled = (tam.config.TargetAllocator != nil) && isPodNameAvailable()
+	if tam.enabled {
+		tam.loadManager(logLevel)
+	}
+	return &tam
+}
+func (tam *TargetAllocatorManager) loadManager(logLevel *promlog.AllowedLevel) {
+	receiverSettings := receiver.Settings{
+		ID: component.MustNewID(strings.ReplaceAll(tam.config.TargetAllocator.CollectorID, "-", "_")),
+		TelemetrySettings: component.TelemetrySettings{
+			Logger:         createLogger(logLevel),
+			TracerProvider: nil,
+			MeterProvider:  nil,
+			MetricsLevel:   0,
+			Resource:       pcommon.Resource{},
+			ReportStatus:   nil,
+		},
+	}
+
+	tam.manager = tamanager.NewManager(receiverSettings, tam.config.TargetAllocator, (*promconfig.Config)(tam.config.PrometheusConfig), false)
+}
+func (tam *TargetAllocatorManager) loadConfig(filename string) error {
+	config, err := loadConfigFromFilename(filename)
+	if err != nil {
+		return err
+	}
+	tam.config = config
+	if tam.config.TargetAllocator == nil {
+		return nil // no target allocator return
+	}
+	//has target allocator
+	tam.config.TargetAllocator.TLSSetting.CAFile = DEFAULT_TLS_CA_FILE_PATH
+	tam.config.TargetAllocator.TLSSetting.ReloadInterval = DEFAULT_TLS_RELOAD_INTERVAL_SECONDS
+	return nil
+}
+func (tam *TargetAllocatorManager) Run() error {
+	err := tam.manager.Start(context.Background(), tam.host, tam.sm, tam.dm)
+	if err != nil {
+		return err
+	}
+	err = tam.reloadConfigTicker()
+	if err != nil {
+		close(tam.shutdownCh)
+		return err
+	}
+	// go ahead and let dependencies know TA is ready
+	close(tam.taReadyCh)
+	//don't stop until shutdown
+	<-tam.shutdownCh
+	return nil
+}
+func (tam *TargetAllocatorManager) Shutdown() {
+	tam.manager.Shutdown()
+	close(tam.shutdownCh)
+}
+func (tam *TargetAllocatorManager) AttachReloadConfigHandler(handler func(config *promconfig.Config)) {
+	tam.reloadConfigHandler = handler
+}
+func (tam *TargetAllocatorManager) reloadConfigTicker() error {
+	if tam.config.TargetAllocator == nil {
+		return fmt.Errorf("target Allocator is not configured properly")
+	}
+	if tam.reloadConfigHandler == nil {
+		return fmt.Errorf("target allocator reload config handler is not configured properly")
+	}
+	fmt.Printf("Starting Target Allocator Reload Config Ticker with %fs interval\n", tam.config.TargetAllocator.Interval.Seconds())
+	ticker := time.NewTicker(tam.config.TargetAllocator.Interval)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				tam.reloadConfigHandler((*promconfig.Config)(tam.config.PrometheusConfig))
+			case <-tam.shutdownCh:
+				ticker.Stop()
+				// Stop the ticker and exit when stop is signaled
+				fmt.Printf("Stopping Target Allocator Reload Config Ticker \n")
```

Comment:
nit: I see we have a mix of `level.Log` and `fmt.Printf` in this file. Which one do we want to use?

Reply:
Right now, I only have access to the logger when I first create the manager. Thats why I use both of them I am open to changing though